### ISSUE TITLE
Script error in PlayerWatchesRoundWinningKillReplay(): Attempted to call GetEncodedEHandle on invalid entity

### ISSUE THREAD

**Issue by fvnkhead** [R2Northstar/NorthstarMods#239]
Server crashed with this while hosting a gun game (Northstar 1.5.1):

```
[17:12:41] [info] [SERVER SCRIPT] SCRIPT ERROR: [SERVER] Attempted to call GetEncodedEHandle on invalid entity
[17:12:41] [info] [SERVER SCRIPT]  -> player.SetKillReplayInflictorEHandle( attacker.GetEncodedEHandle() )
[17:12:41] [info] [SERVER SCRIPT]
CALLSTACK
*FUNCTION [PlayerWatchesRoundWinningKillReplay()] mp/_gamestate_mp.nut line [380]

[17:12:41] [info] [SERVER SCRIPT] LOCALS
[attacker] ENTITY (NULL)
[replayLength] 7.5
[player] ENTITY (player erri120-origin [1] (player "erri120-origin" at <-80.4129 1772.69 1.03125>))
[this] TABLE

DIAGPRINTS

[17:12:41] [info] [SERVER SCRIPT] SCRIPT ERROR: [SERVER] Attempted to call GetEncodedEHandle on invalid entity
[17:12:41] [info] [SERVER SCRIPT]  -> player.SetKillReplayInflictorEHandle( attacker.GetEncodedEHandle() )
[17:12:41] [info] [SERVER SCRIPT]
CALLSTACK
*FUNCTION [PlayerWatchesRoundWinningKillReplay()] mp/_gamestate_mp.nut line [380]

[17:12:41] [info] [SERVER SCRIPT] LOCALS
[attacker] ENTITY (NULL)
[replayLength] 7.5
[player] ENTITY (player FattChicken4u [2] (player "FattChicken4u" at <230.735 -1450.11 1.03125>))
[this] TABLE

DIAGPRINTS

[17:12:41] [info] [SERVER SCRIPT] SCRIPT ERROR: [SERVER] Attempted to call GetEncodedEHandle on invalid entity
[17:12:41] [info] [SERVER SCRIPT]  -> player.SetKillReplayInflictorEHandle( attacker.GetEncodedEHandle() )
[17:12:41] [info] [SERVER SCRIPT]
CALLSTACK
*FUNCTION [PlayerWatchesRoundWinningKillReplay()] mp/_gamestate_mp.nut line [380]

[17:12:41] [info] [SERVER SCRIPT] LOCALS
[attacker] ENTITY (NULL)
[replayLength] 7.5
[player] ENTITY (player Neinguar [3] (player "Neinguar" at <1006.05 -1130.53 1.03125>))
[this] TABLE

DIAGPRINTS

[17:12:41] [info] [SERVER SCRIPT] SCRIPT ERROR: [SERVER] Attempted to call GetEncodedEHandle on invalid entity
[17:12:41] [info] [SERVER SCRIPT]  -> player.SetKillReplayInflictorEHandle( attacker.GetEncodedEHandle() )
[17:12:41] [info] [SERVER SCRIPT]
CALLSTACK
*FUNCTION [PlayerWatchesRoundWinningKillReplay()] mp/_gamestate_mp.nut line [380]

[17:12:41] [info] [SERVER SCRIPT] LOCALS
[attacker] ENTITY (NULL)
[replayLength] 7.5
[player] ENTITY (player Rrebetol [5] (player "Rrebetol" at <741.766 2217.74 96.0313>))
[this] TABLE

DIAGPRINTS

[17:12:41] [info] [SERVER SCRIPT] SCRIPT ERROR: [SERVER] Attempted to call GetEncodedEHandle on invalid entity
[17:12:41] [info] [SERVER SCRIPT]  -> player.SetKillReplayInflictorEHandle( attacker.GetEncodedEHandle() )
[17:12:41] [info] [SERVER SCRIPT]
CALLSTACK
*FUNCTION [PlayerWatchesRoundWinningKillReplay()] mp/_gamestate_mp.nut line [380]

[17:12:41] [info] [SERVER SCRIPT] LOCALS
[attacker] ENTITY (NULL)
[replayLength] 7.5
[player] ENTITY (player XyzoDude [6] (player "XyzoDude" at <340.223 1653.36 1.03125>))
[this] TABLE

DIAGPRINTS

[17:12:41] [info] [SERVER SCRIPT] SCRIPT ERROR: [SERVER] Attempted to call GetEncodedEHandle on invalid entity
[17:12:41] [info] [SERVER SCRIPT]  -> player.SetKillReplayInflictorEHandle( attacker.GetEncodedEHandle() )
[17:12:41] [info] [SERVER SCRIPT]
CALLSTACK
*FUNCTION [PlayerWatchesRoundWinningKillReplay()] mp/_gamestate_mp.nut line [380]

[17:12:41] [info] [SERVER SCRIPT] LOCALS
[attacker] ENTITY (NULL)
[replayLength] 7.5
[player] ENTITY (player PleaseNotThisOne [7] (player "PleaseNotThisOne" at <-815.306 -563.717 1.03125>))
[this] TABLE

DIAGPRINTS

[17:12:41] [info] [SERVER SCRIPT] SCRIPT ERROR: [SERVER] Attempted to call GetEncodedEHandle on invalid entity
[17:12:41] [info] [SERVER SCRIPT]  -> player.SetKillReplayInflictorEHandle( attacker.GetEncodedEHandle() )
[17:12:41] [info] [SERVER SCRIPT]
CALLSTACK
*FUNCTION [PlayerWatchesRoundWinningKillReplay()] mp/_gamestate_mp.nut line [380]

[17:12:41] [info] [SERVER SCRIPT] LOCALS
[attacker] ENTITY (NULL)
[replayLength] 7.5
[player] ENTITY (player The_zucc_10 [8] (player "The_zucc_10" at <434.149 2156.4 1.03125>))
[this] TABLE

DIAGPRINTS

[17:12:41] [info] [SERVER SCRIPT] SCRIPT ERROR: [SERVER] Attempted to call GetEncodedEHandle on invalid entity
[17:12:41] [info] [SERVER SCRIPT]  -> player.SetKillReplayInflictorEHandle( attacker.GetEncodedEHandle() )
[17:12:41] [info] [SERVER SCRIPT]
CALLSTACK
*FUNCTION [PlayerWatchesRoundWinningKillReplay()] mp/_gamestate_mp.nut line [380]

[17:12:41] [info] [SERVER SCRIPT] LOCALS
[attacker] ENTITY (NULL)
[replayLength] 7.5
[player] ENTITY (player sl1deshow_gaming [9] (player "sl1deshow_gaming" at <1188.63 821.087 1.03125>))
[this] TABLE

DIAGPRINTS

[17:12:41] [info] [SERVER SCRIPT] SCRIPT ERROR: [SERVER] Attempted to call GetEncodedEHandle on invalid entity
[17:12:41] [info] [SERVER SCRIPT]  -> player.SetKillReplayInflictorEHandle( attacker.GetEncodedEHandle() )
[17:12:41] [info] [SERVER SCRIPT]
CALLSTACK
*FUNCTION [PlayerWatchesRoundWinningKillReplay()] mp/_gamestate_mp.nut line [380]

[17:12:41] [info] [SERVER SCRIPT] LOCALS
[attacker] ENTITY (NULL)
[replayLength] 7.5
[player] ENTITY (player f09b0e9db4aba1342b4936bfef21b7a7 [12] (player "f09b0e9db4aba1342b4936bfef21b7a7" at <565.762 -1618.77 1.03125>))
[this] TABLE

DIAGPRINTS

[17:12:41] [info] [SERVER SCRIPT] [ParseableLog]{"subject":"round","verb":"ended"}
[17:12:41] [error] Northstar has crashed! a minidump has been written and exception info is available below:
[17:12:41] [error] Cause: Access Violation
Attempted to read from: 0x00000000
[17:12:41] [error] At: engine.dll + 0x247d3a
[17:12:41] [error]     Northstar.dll + 0x128ea47a
[17:12:41] [error]     ntdll.dll + 0x17002567d
[17:12:41] [error]     ntdll.dll + 0x1700548c5
[17:12:41] [error]     ntdll.dll + 0x17005245e
[17:12:41] [error]     engine.dll + 0x18677d3a
[17:12:41] [error]     engine.dll + 0x18847466
[17:12:41] [error]     engine.dll + 0x18550c2f
[17:12:41] [error]     engine.dll + 0x18550542
[17:12:41] [error]     engine.dll + 0x1859db55
[17:12:41] [error]     engine.dll + 0x185f88c0
[17:12:41] [error]     Northstar.dll + 0x128d96d6
[17:12:41] [error]     engine.dll + 0x185f652f
[17:12:41] [error]     engine.dll + 0x187c0e2d
[17:12:41] [error]     engine.dll + 0x185f79d9
[17:12:41] [error]     engine.dll + 0x185f793c
[17:12:41] [error]     launcher.dll + 0x12b05afd
[17:12:41] [error]     launcher.dll + 0x12b05afd
[17:12:41] [error]     launcher.dll + 0x12afd386
[17:12:41] [error]     NorthstarLauncher.exe + 0x140003e7d
[17:12:41] [error]     NorthstarLauncher.exe + 0x140007000
[17:12:41] [error]     kernel32.dll + 0x7b62c599
[17:12:41] [error]     ntdll.dll + 0x170059d53
[17:12:41] [error]     NorthstarLauncher.exe + 0x0
[17:12:41] [error]     NorthstarLauncher.exe + 0x140007070
[17:12:41] [error]     NorthstarLauncher.exe + 0x67ff0000
[17:12:41] [error]     NorthstarLauncher.exe + 0x0
[17:12:41] [error] RAX: 0x0
[17:12:41] [error] RBX: 0x18c156d0
[17:12:41] [error] RCX: 0x0
[17:12:41] [error] RDX: 0x0
[17:12:41] [error] RSI: 0x2b50bb78
[17:12:41] [error] RDI: 0x2b50bb78
[17:12:41] [error] RBP: 0x2b50bd88
[17:12:41] [error] RSP: 0x7cf470
[17:12:41] [error] R8: 0x2b50bb78
[17:12:41] [error] R9: 0x65
[17:12:41] [error] R10: 0x2b50bd94
[17:12:41] [error] R11: 0x18a378ec
[17:12:41] [error] R12: 0x0
[17:12:41] [error] R13: 0x0
[17:12:41] [error] R14: 0xffffffff
[17:12:41] [error] R15: 0x1
0024:fixme:dbghelp_dwarf:dwarf2_read_range no entry found
0024:fixme:dbghelp_dwarf:dwarf2_read_range no entry found
0024:fixme:dbghelp:MiniDumpWriteDump NIY MiniDumpScanMemory
Error: Failed to run server: server exited: exit status 100.
```

**Comment by RomeoCantCode** [comment]
@fvnkhead could it be that this happens, when the winning player leaves before winning kill cam is played? are you able to reproduce that?

**Comment by fvnkhead** [comment]
@RomeoCantCode I don't have time to test this at the moment, but I got this DM:

```
i believe i may know what is responsible for this. i noticed one day i left right before my killcam played, and ~10 minutes later i was able to get back on and the server was at 0 players. this happened a few more times over the weekend. i’m not fully sure if these events are related but i was just letting you know as it may be an explanation.
```

So it looks like that might be the case

**Comment by ASpoonPlaysGames** [comment]
Stale, but also probably fixed by #516
